api_client/python/timesketch_api_client/importer.py

```diff
@@ -51,7 +63,54 @@ def __init__(self, entry_threshold=None):
         self._timeline_name = None
         self._timestamp_desc = None
 
-        self._threshold = entry_threshold or self.ENTRY_THRESHOLD
+        self._chunk = 1
+
+        self._text_encoding = self.DEFAULT_TEXT_ENCODING
+        self._threshold_entry = self.DEFAULT_ENTRY_THRESHOLD
+        self._threshold_filesize = self.DEFAULT_FILESIZE_THRESHOLD
+
+    def _fix_dict(self, my_dict):
+        """Adjusts a dict with so that it can be uploaded to Timesketch.
+
+        Args:
+            my_dict: a dictionary that may be missing few fields needed
+                    for Timesketch.
+        """
+        if 'message' not in my_dict and self._format_string:
+            my_dict['message'] = self._format_string.format(**my_dict)
+
+        _ = my_dict.setdefault('timestamp_desc', self._timestamp_desc)
+
+        if 'datetime' not in my_dict:
+            for key in my_dict:
+                key_string = key.lower()
+                if 'time' not in key_string:
+                    continue
+
+                if key_string == 'timestamp_desc':
+                    continue
+
+                value = my_dict[key]
+                if isinstance(value, six.string_types):
+                    try:
+                        date = dateutil.parser.parse(my_dict[key])
+                        my_dict['datetime'] = date.isoformat()
+                        break
+                    except ValueError:
+                        continue
+                else:
+                    try:
+                        date = datetime.datetime.utcfromtimestamp(value / 1e6)
+                        my_dict['datetime'] = date.isoformat()
+                        break
+                    except ValueError:
+                        continue
+
+        # We don't want to include any columns that start with an underscore.
+        underscore_columns = [x for x in my_dict if x.startswith('_')]
```

Comment:
I guess this is due to the "special" TS fields, like __ts_label ? 

I wonder if we will miss anything else that might have underscore in the name?

Reply:
this is due to those kind of field names. Do you have suggestions? Have a disclaimer?

Reply:
done